bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/async/AbstractRetryingOperation.java

```diff
@@ -321,14 +322,14 @@ protected void run() {
 
   protected CallOptions getCallOptions() {
     if (callOptions.getDeadline() != null) {
-      return callOptions;
-    }
-    if (request instanceof ReadRowsRequest
-        && !CallOptionsFactory.ConfiguredCallOptionsFactory.isGet((ReadRowsRequest) request)) {
-      // This is a streaming read.
-      return callOptions;
+      return getUserCallOptions();
+    } else {
+      return getUserCallOptions().withDeadlineAfter(UNARY_DEADLINE_MINUTES, TimeUnit.MINUTES);
```

Comment:
Can we make this change in CallOptionsFactory.java.create()? Setting this deadline in retry logic seems weird.